### ISSUE TITLE
Warehouse state refresh after alter sets statement_timeout_in_seconds to 0 incorrectly

### ISSUE THREAD

**Issue by Daniel-DeMara** [chanzuckerberg/terraform-provider-snowflake#601]
25.11 <--The provider version you are using.

14.3 <--The version of Terraform you were using when the bug was encountered.

When running terraform plan and apply, existing warehouses are flagged to have 'statement_timeout_in_seconds' changed from 0 -> (our set value) on concurrent runs.  

I have discovered that in snowflake when you run 'SHOW WAREHOUSE <NAME>', statement_timeout_in_seconds is not a returned attribute and likely defaults to 0 when that column is not returned. In order to retrieve the actual statement_timeout_in_seconds the snowflake command would be 'SHOW PARAMETERS FOR WAREHOUSE <NAME>'

When following what commands are run on snowflake, the ALTER on the warehouse is complete, and then a follow up SHOW is ran on the same warehouse to seemingly verify the command, this is likely where the state file is updated with a 0 value, for a subsequent run to realize our terraform code has this setting > 0 causing it to be altered again. 

I see that the file /pkg/resources/warehouse.go has a function ReadWarehouse where it only implements the Show command on the warehouse. 

I would expect terraform to pull the correct value for statement_timeout_in_seconds 

This issue was resolved by rolling back the provider version to a fixed version of 25.10.

Thank you very much for your time!

**Comment by alldoami** [comment]
Are you saying that even after you apply to your new set value for `statement_timeout_in_seconds`, it still says it is `0`?

**Comment by TonyGaul** [comment]
Hi, in my case I have this same problem.
In TF I use: statement_timeout_in_seconds = 3600
When TF first runs it sets the Snowflake parameter "STATEMENT_TIMEOUT_IN_SECONDS" to 3600
I checked the SF param
When TF runs a second time I get: ~ statement_timeout_in_seconds        = 0 -> 3600

In all TF runs, the Snowflake params are:
STATEMENT_QUEUED_TIMEOUT_IN_SECONDS = 0 
STATEMENT_TIMEOUT_IN_SECONDS =3600

**Comment by Daniel-DeMara** [comment]
@alldoami Yes, Terraform attempts to apply the same change repeatedly, so it is not successfully setting the state file to the correct value. It's also not necessarily a new value, this was old infrastructure that was in place prior to 25.11 being released.

**Comment by yonieilon** [comment]
I'm seeing the same behaviour in version `0.25.12` as well when I just tried to upgrade provider version in our project...

**Comment by avifried1** [comment]
happening in version `0.25.18` as well, TF version is `1.0.5`

**Comment by sonmezonur** [comment]
#745  should fix this issue. Could you please try the latest versions?
